### PR TITLE
#77 Swarm.leave + unit tests

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #77 
Implemented Swarm.leave and added unit tests. Left puzzle for continuing.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#90](https://github.com/amihaiemil/docker-java-api/issues/90) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#90](https://github.com/amihaiemil/docker-java-api/issues/90) is assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/1151872945); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/test/java/com/amihaiemil/docker/RtSwarmTestCase.java [hunk 1]
```diff
@@ -44,6 +44,95 @@
...
+    @Test
+    public void forcedLeave() throws Exception {
+        new RtSwarm(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_OK,
+                    "{}"
+                ),
+                new Condition(
+                    "Request method should be GET",
```
Comment (llorllale) [inline src/test/java/com/amihaiemil/docker/RtSwarmTestCase.java]: @amihaiemil typo here

File: src/test/java/com/amihaiemil/docker/RtSwarmTestCase.java [hunk 2]
```diff
@@ -44,6 +44,95 @@
...
+    @Test
+    public void leavesWithoutForce() throws Exception {
+        new RtSwarm(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_OK,
+                    "{}"
+                ),
+                new Condition(
+                    "Request method should be GET",
```
Comment (llorllale) [inline src/test/java/com/amihaiemil/docker/RtSwarmTestCase.java]: @amihaiemil typo here

**Comment by llorllale** [comment]
@amihaiemil just two minor typos to fix then we're golden

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@009e3d45-fe1c-4664-83fc-70da2c75bbad, thread: AsyncFlush-4
com.zerocracy.farm.reactive.StkGroovy[125] java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/cost/pay_architect_for_pull_request.groovy
sun.reflect.GeneratedMethodAccessor69[-1] java.lang.reflect.InvocationTargetException: null
com.jcabi.xml.DomParser[112] java.lang.IllegalArgumentException: Invalid XML: "<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
"
com.sun.org.apache.xerces.internal.parsers.DOMParser[257] org.xml.sax.SAXParseException: The element type "hr" must be terminated by the matching end-tag "</hr>".
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm)

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@009e3d45-fe1c-4664-83fc-70da2c75bbad, thread: AsyncFlush-3
com.zerocracy.farm.reactive.StkGroovy[125] java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pmo/awards/reject_people_below_score.groovy
sun.reflect.GeneratedMethodAccessor213[-1] java.lang.reflect.InvocationTargetException: null
com.jcabi.xml.DomParser[112] java.lang.IllegalArgumentException: Invalid XML: "<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
"
com.sun.org.apache.xerces.internal.parsers.DOMParser[257] org.xml.sax.SAXParseException: The element type "hr" must be terminated by the matching end-tag "</hr>".
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm)

**Comment by 0crat** [comment]
The job [#90](https://github.com/amihaiemil/docker-java-api/issues/90) is now out of scope

**Comment by 0crat** [comment]
The job is not in WBS, won't close the order

**Pull request merged**